docker/final.Dockerfile

```diff
@@ -3,8 +3,13 @@
 # Final layer containing all artifacts.
 FROM debian:bookworm-slim
 
+# Allow apt to retry 3 times before failing.
 RUN echo 'Acquire::Retries "3";' > /etc/apt/apt.conf.d/80-retries
 
+# Force apt to timeout retrieval of data after 3 minutes.
+RUN echo 'Acquire::http::timeout "180";' > /etc/apt/apt.conf.d/99-timeouts
+RUN echo 'Acquire::https::timeout "180";' >> /etc/apt/apt.conf.d/99-timeouts
+
```

Comment:
If you add the following to the Dockerfile, you can make the shell used by `RUN` use `-x` debug output which should help identify the command that hangs.
```
SHELL ["/bin/sh", "-c", "-x"] 
```

Additionally may make sense to place the following into the task's environment so buildkit doesn't collapse output, too:
```
BUILDKIT_PROGRESS=plain
```

Reply:
Added those. The new logs in https://cirrus-ci.com/task/5244127424544768 look better.